src/thunderbots/software/grsim_communication/grsim_backend.h

```diff
@@ -3,14 +3,14 @@
 #include <boost/asio.hpp>
 #include <string>
 
+#include "ai/primitive/primitive.h"
 #include "ai/world/team.h"
 #include "geom/angle.h"
 #include "geom/point.h"
 #include "proto/grSim_Packet.pb.h"
-#include "robot_communication/backend.h"
 
 
-class GrSimBackend : public Backend
+class GrSimBackend
```

Comment:
I removed the generic `Backend` interface because I think the `grsim` and `radio` nodes are going to be different enough it doesn't make sense. Particularly in the case of the `sendPrimitives` function.

For example, the `grsim` node's `sendPrimitives` function will also need to accept the current state of the Friendly Team, because that information is required to translate coordinates and simulate the motion of the robots, since that takes place in the node.

On the other hand, the `radio` node does not need this information. Because the control is performed on the robot, the `sendPrimitives` function only needs the list of `Primitives` to send.

Of course each node will also have its own specific functions. For example, the `radio` node will likely have a separate function that just sends the "vision threading" information, which is the robot and ball positions on the field. This is not required for the `grsim` node. This doesn't affect the inheritance from the generic `Backend`, but I figured it was worth mentioning for completeness.

@henrymliu to correct me if I misspoke about the radio interface.

Reply:
Great explanation, thank you! This seems reasonable, particularly because the interface is now enforced by ROS message type now.